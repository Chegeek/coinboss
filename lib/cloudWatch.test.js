import AWS from 'aws-sdk-mock';
import CloudWatch from './cloudWatch';

describe('CloudWatch', () => {
  describe('setupBuySellDimensions()', () => {
    test('returns an array of dimensions', () => {
      expect(CloudWatch.setupBuySellDimensions('ETH', 'EUR')).toMatchSnapshot();
    });
  });

  describe('putPriceMetric()', () => {
    beforeEach(() => AWS.mock('CloudWatch', 'putMetricData', { status: 'mocked' }));
    afterEach(() => AWS.restore('CloudWatch'));

    test('stores metrics within CloudWatch', () =>
      new CloudWatch().putPriceMetric({
        metricName: 'foo',
        currency: 'ETH',
        localCurrency: 'EUR',
        price: '0.19',
      })
        .then(data => expect(data.status).toBe('mocked')),
    );
  });
});
