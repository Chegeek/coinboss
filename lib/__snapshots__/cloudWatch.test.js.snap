// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudWatch setupBuySellDimensions() returns an array of dimensions 1`] = `
Array [
  Object {
    "Name": "CryptoCurrency",
    "Value": "ETH",
  },
  Object {
    "Name": "LocalCurrency",
    "Value": "EUR",
  },
  Object {
    "Name": "Stage",
    "Value": "dev",
  },
]
`;
